### PR TITLE
#160283233 set up axios instance

### PR DESCRIPTION
**Pull request by tutugodfrey**
#### What does this PR do?
- Add axios for making API requests

#### Description of Task to be completed?
- Install axios as project dependencies
- Set up and axios instance
- Add a function for making API requests using axios

#### How should this be manually tested?
- Clone the repository and in your terminal `npm install` to install dependencies, then run `npm run dev` to start the application in development mode
- When the browser open try clicking the button `Secondary` and check the console for messages

#### Any background context you want to provide?
- Two functions are defined one is for starting an axios instances which can be use out of the box
- The function apiCalls is accepts the axios instance, and other parameters (header method, body) and make an api call using axios

#### What are the relevant pivotal tracker stories?
#160283233

#### Screenshots (if appropriate)
image
image

#### Questions:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,14 +20,21 @@
...
     "semantic-ui-react": "^0.82.3"
   },
   "devDependencies": {
+    "@babel/core": "^7.0.0",
+    "@babel/plugin-transform-runtime": "^7.0.0",
+    "@babel/preset-env": "^7.0.0",
+    "@babel/preset-react": "^7.0.0",
     "babel": "^6.23.0",
-    "babel-core": "^6.26.3",
+    "babel-core": "^7.0.0-bridge.0",
```
Comment (rhenshaw56) [inline package.json]: @tutugodfrey why the change ?

File: src/App.js [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
 /**
  *This class is the main entry point for the application
  */
 class App extends Component {
+  /**
+   * example function of how to make http request
+   * using axios instance
+   * @returns { Object } response from server
+   */
+  makeRequest = async () => {
```
Comment (rhenshaw56) [inline src/App.js]: you can't do this, do it right, cos elint is flagging you

File: src/services/requestHandler.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import axios from 'axios';
+
+/**
+ * This function creates an instance of axios and return
+ * it for making API request
+ * @param { Object} headers an object of http headers
+ * @param { Object } body the request body
+ * @returns { instance } an instance of axios
+ */
+const axiosInstance = (headers, body) => {
```
Comment (rhenshaw56) [inline src/services/requestHandler.js]: add the default headers to the instance, we don't want to be passing in headers every time we need to make a request

File: src/services/requestHandler.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+ * @param {*} axiosInStance an instance of axios
+ * @param { String } method http method e.g post, get, put, delete
+ * @param { String } path the relative path of the request
+ * @param { Object } headers an object http headers
+ * @param { Object } body the request body
+ * required for making post and put request
+ * @returns { Object } response the response from the server
+ */
+const apiCalls = async (axiosInStance, method, path, headers, body) => {
+  /* eslint-disable no-console */
```
Comment (rhenshaw56) [inline src/services/requestHandler.js]: take all this out, you don't need it, it just complicates things

**Review by rhenshaw56 — CHANGES_REQUESTED** [review]

File: src/services/requestHandler.js (hunk 4 above)
Comment (tutugodfrey) [reply src/services/requestHandler.js]: Ok, I will attend to them.

**Review by megame24 — APPROVED** [review]

**Review by JCanaks — APPROVED** [review]

**Comment by andela-moe** [comment]
@tutugodfrey I am interested in knowing why the commit to the branch in this PR was done by Joseph Cobhams https://github.com/andela/rivendell-ah-client/pull/10/commits

@andela-moe the mixup is because my system is bad and I was given another system to use for the main time. The username reflect the username of the person formally in possession of the system. But I have updated the information now.

File: src/services/requestHandler.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import axios from 'axios';
+
+/**
+ * This function creates an instance of axios and return
+ * it for making API request
+ * @returns { instance } an instance of axios
+ */
+const axiosInstance = () => {
+  let apiUrl = 'https://authorhaven.herokuapp.com/api';
```
Comment (chidioguejiofor) [inline src/services/requestHandler.js:9]: Instead of doing this. What do you think about using an environmental variable like process.env.BASE_URL

**Review by rhenshaw56 — APPROVED** [review]
